src/main/java/newbank/server/NewBankClientHandler.java

```diff
@@ -6,46 +6,128 @@
 import java.io.PrintWriter;
 import java.net.Socket;
 
+import newbank.server.exceptions.DuplicateCustomerException;
+import newbank.server.exceptions.PasswordInvalidException;
+import newbank.server.exceptions.RequestNotAllowedException;
+
+/** The NewBankClientHandler handles all clients requests. */
 public class NewBankClientHandler extends Thread {
 
   private NewBank bank;
   private BufferedReader in;
   private PrintWriter out;
+  private CustomerID customer;
 
   public NewBankClientHandler(Socket s) throws IOException {
     bank = NewBank.getBank();
     in = new BufferedReader(new InputStreamReader(s.getInputStream()));
     out = new PrintWriter(s.getOutputStream(), true);
   }
 
+  private void handleLogin(String[] tokens) {
+    String username = "";
+    String password = "";
+
+    if (tokens.length >= 2) {
+      username = tokens[1];
+    }
+    if (tokens.length >= 3) {
+      password = tokens[2];
+    }
+
+    customer = bank.checkLogInDetails(username, password);
+
+    if (customer != null) {
+      out.println("SUCCESS: Log In Successful");
+    } else {
+      out.println("FAIL: Log In Failed");
+    }
+  }
+
+  private void handleRegisterCustomer(String[] tokens) {
+    String username = "";
+    String password = "";
+
+    if (tokens.length >= 2) {
+      username = tokens[1];
+    }
+    if (tokens.length >= 3) {
+      password = tokens[2];
+    }
+
+    try {
+      bank.addCustomer(username, password);
+
+      out.println("SUCCESS: Account created successfully.");
+    } catch (DuplicateCustomerException ex) {
+      out.println("FAIL: " + String.format("FAIL: Customer name [%s] already exists.", username));
+    } catch (PasswordInvalidException e) {
+      out.println("FAIL: Specified password does not meet the security requirements.");
+    }
+  }
+
+  private void handleShowAccounts() throws RequestNotAllowedException {
+    checkLoggedIn();
+    out.println(bank.showAccountsFor(customer));
+  }
+
+  private void checkLoggedIn() throws RequestNotAllowedException {
+    if (customer == null) {
+      throw new RequestNotAllowedException();
+    }
+  }
+
+  private boolean processRequest(final String request) {
+    final String[] tokens = request.split("\\s+");
+
+    assert (tokens.length > 0);
+
+    try {
+      switch (tokens[0].toUpperCase()) {
```

Comment:
It's surely an UX improvement, but also a little bit of a global change to the protocol. Perhaps we shall document it.